web/src/HUD.tsx

```diff
@@ -167,6 +182,25 @@ class HUD extends Component<HudProps, HudState> {
       .then(err => console.error(err))
   }
 
+  sendSnapshot(snapshot: Snapshot) {
+    let url = `//${window.location.host}/api/snapshot/new`
+    fetch(url, {
+      method: "post",
+      body: JSON.stringify(snapshot),
+    })
+      .then(res => {
```

Comment:
I think this promise could be re-written more idiomatically as:

```typescript
fetch()
  .then(res => res.json)
  .then((value: NewSnapshotResponse) => {
            this.setState({
              SnapshotLink: value.url,
            })
  })
  .catch(err => console.error(err)
```

then the same catch will handle everything.